### ISSUE TITLE
Donate form QA: Update field labels and add placeholders to SEPA form

### ISSUE THREAD

**Issue by flukeout** [mozilla/donate.mozilla.org#1876]
We may want to update the labels, and maybe add placeholders or helper text for these fields...

![image](https://user-images.githubusercontent.com/25212/32745382-68ec578a-c867-11e7-94e4-9a86129abbcd.png)

* **Name** should probably be changed to something like **Account Holder's Full Name**
  * @gideonthomas is there a spec for this somewhere?
* Do we want to add some helper text for the IBAN number? Like, what it stands for or how to figure out which one to use? 
  * Thinking of things like the popups / extra info that help you find your 3-digit security number on credit cards

**Comment by gideonthomas** [comment]
> Name should probably be changed to something like Account Holder's Full Name

I looked at a few donation forms that accept SEPA (not-so-great ones). The kinds of text they have are "Debtor's Name", "Name of account holder(s)", and "Account Holder Name(s)". I don't think there is a spec anywhere for what it should be called.

> Do we want to add some helper text for the IBAN number? Like, what it stands for or how to figure out which one to use?

The IBAN is the account number so I'm not sure what help text to show. The other donation forms I looked at had the label be something along the lines of "Account number (IBAN)". There are some rules for validating IBANs (for e.g. up to 34 characters) but I don't think that should be in the help text. We might want to give an example maybe? Maybe placeholder text like "for e.g. DE44 5001 0517 5407 3249 31"?

According to Wikipedia, "When printed it is expressed in groups of four characters separated by a single space, the last group being of variable length". Maybe we want to print those spaces as the user types in the number (but the actual number we use in code must not have any spaces)? Def. not a super high-priority though.
